examples/drag_and_drop/src/lib.rs

```diff
@@ -0,0 +1,198 @@
+                "Drag me and let's go!"
+            } else {
+                "Put mouse on me"
+            };
+            let draggable = enter;
+            div_drag.set_inner_text(text);
+            div_drag.set_draggable(draggable);
+        }
+        Msg::Dragging(id) => {
+            let div_drag = get_div_from_id(id, &model);
+            div_drag.set_inner_text("OUUUUUUUUUUUUUIII");
+            model.who_is_getting_dragged = Some(id);
+        }
+        Msg::DragEnded(id) => {
+            let div_drag = get_div_from_id(id, &model);
+
+            if let Some(container) = div_drag.parent_element() {
+                if container.id() != "drop_zone" {
+                    div_drag.set_inner_text("NOO :(");
+                } else {
+                    div_drag.set_inner_text("Now I am happy");
+                }
+            }
+
+            model.who_is_getting_dragged = None
+        }
+        Msg::DragOver => {}
+    }
+}
+
+fn get_div_from_id(id: u8, model: &Model) -> HtmlDivElement {
+    if id == 1 {
+        model.drag_me_1.get().expect("should get div 1")
+    } else if id == 2 {
+        model.drag_me_2.get().expect("should get div 2")
+    } else {
+        panic!("wrong id");
+    }
+}
+// ------ ------
+//     View
+// ------ ------
+
+// Note: It's macro so you can use it with all events.
+macro_rules! stop_and_prevent {
+    { $event:expr } => {
+        {
+            $event.stop_propagation();
+            $event.prevent_default();
+        }
+     };
+}
+trait IntoDragEvent {
+    fn into_drag_event(self) -> DragEvent;
+}
+
+impl IntoDragEvent for Event {
+    fn into_drag_event(self) -> DragEvent {
+        self.dyn_into::<web_sys::DragEvent>()
+            .expect("cannot cast given event into DragEvent")
+    }
+}
+
+fn view(model: &Model) -> Node<Msg> {
+    div![
+        generate_draggable_div(2, "I am number1", &model.drag_me_2, "green"),
+        generate_draggable_div(1, "I am number1", &model.drag_me_1, "purple"),
+        div![
+            id!("drop_zone"),
+            el_ref(&model.drop_zone),
+            "drop stuff there",
+            style![
+                St::Height => px(150),
+                St::Width => px(150),
+                St::Margin => "auto",
+                St::FontFamily => "sans-serif",
+                St::Display => "flex",
+                St::FlexDirection => "column",
+                St::TextAlign=> "center",
+                St::JustifyContent => "center",
+                St::AlignItems => "center",
+                St::Border => [&px(2), "dotted", "green"].join(" ");
+                St::BorderRadius => px(20),
+            ],
+            ev(Ev::Drop, |_| { Msg::Drop }),
+            ev(Ev::DragOver, |event| {
+                let drag_event = event.into_drag_event();
+                stop_and_prevent!(drag_event);
+                drag_event.data_transfer().unwrap().set_drop_effect("move");
+                Msg::DragOver
+            })
```

Comment:
Could we use https://github.com/seed-rs/seed/blob/master/src/browser/dom/event_handler.rs#L91 or at least https://github.com/seed-rs/seed/blob/master/src/browser/dom/cast.rs#L58 ?

Reply:
Done. I did not know we had this available. We need to update our dropzone example as well :wink: 

https://github.com/seed-rs/seed/blob/master/examples/drop_zone/src/lib.rs